src/main/java/com/github/timm/cucumber/generate/CucumberITGenerator.java

```diff
@@ -1,226 +1,238 @@
-                    config.getCucumberOutputDir()
-                    .replace('\\', '/'), fileCounter, formatStr));
-
-            if (i < formatStrs.length - 1) {
-                sb.append(", ");
-            }
-        }
-        return sb.toString();
-    }
-
-    /**
-     * Wraps each package in quotes for use in the template
-     */
-    private String quoteGlueStrings() {
-        final String[] packageStrs = overriddenParameters.getGlue().split(",");
-
-        final StringBuilder sb = new StringBuilder();
-
-        for (int i = 0; i < packageStrs.length; i++) {
-            final String packageStr = packageStrs[i];
-            sb.append(String.format("\"%s\"", packageStr.trim()));
-
-            if (i < packageStrs.length - 1) {
-                sb.append(", ");
-            }
-        }
-        return sb.toString();
-    }
-
-}
+package com.github.temyers.generate;
+
+import com.github.temyers.options.TagParser;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+public class CucumberITGenerator {
+
+    private final FileGeneratorConfig config;
+    private final OverriddenCucumberOptionsParameters overriddenParameters;
+    private final OverriddenRerunOptionsParameters rerunOptionsParameters;
+    int fileCounter = 1;
+    private String featureFileLocation;
+    private Template velocityTemplate;
+    private ClassNameGenerator classNameGenerator=new ClassNameGenerator();
+    private String outputFileName;
+    private String htmlFormat;
+    private String jsonFormat;
+    private String rerunFormat;
+
+
+
+    public CucumberITGenerator(final FileGeneratorConfig config, final OverriddenCucumberOptionsParameters overriddenParameters,final OverriddenRerunOptionsParameters rerunOptionsParameters ) {
+        this.config = config;
+        this.overriddenParameters = overriddenParameters;
+        this.rerunOptionsParameters= rerunOptionsParameters;
+        initTemplate();
+    }
+
+    private void initTemplate() {
+        final Properties props = new Properties();
+        props.put("resource.loader", "class");
+        props.put("class.resource.loader.class",
+                "org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader");
+        final VelocityEngine engine = new VelocityEngine(props);
+        engine.init();
+        if (config.useTestNG()){
+            velocityTemplate = engine.getTemplate("cucumber-testng-runner.vm",config.getEncoding());
+        }else if (config.useReRun()){
+            velocityTemplate = engine.getTemplate("cucumber-junit-re-runner.vm",config.getEncoding());

```

Comment:
Depending on the discussion above about which properties to support, may need a testng re-runner file (or at least make it explicit that testng rerunning isn't currently supported)